packages/libs/src/sdk/middleware/addRequestSignatureMiddleware.ts

```diff
@@ -27,11 +28,13 @@ export const addRequestSignatureMiddleware = ({
     pre: async (context: RequestContext): Promise<FetchParams> => {
       const { auth, logger } = services
       try {
+        const walletAddress = await auth.getAddress()
         const currentTimestamp = new Date().getTime()
         const isExpired =
           !timestamp || timestamp + SIGNATURE_EXPIRY_MS < currentTimestamp
 
-        if (!message || !signature || isExpired) {
+        if (!message || !signature || isExpired || address !== walletAddress) {
```

Comment:
nit took me a sec to read, 

```
const walletAddressChanged = address !== await auth.getAddress()
if (!message || !signature || isExpired || walletAddressChanged)
```

Reply:
I'm going to compromise halfway in between since we need to access the result of `auth.getAddress()` twice and I don't want to call it twice.